### PR TITLE
chore(trading): 4384 all markets in switcher

### LINKED ISSUE #4384: Default to an "all" markets view in the switcher
## The Chore

While the number of markets on Vega is small, it is better if we present one list to give visitors an overview of all activity.

The Markets page has an "All" tab for this purpose, but the market switcher only has tabs for product types.

Introduce an "All" tab on the market switcher.

## Tasks

- [x] Add an "All" tab to the market switcher, which shows all markets sorted by volume
- [x] Default to the "All" tab
- [ ] When viewing the "All" tab, introduce a label or other cue to distinguish spot, future, and perpetual markets from
- [x] Rename (or possible remove) 'All markets' link as it'll be weird to have it next to the 'All' tab
 each other, for example [pills or icons](https://github.com/vegaprotocol/frontend-monorepo/issues/4319)

## Additional details / background info
- Related: https://github.com/vegaprotocol/frontend-monorepo/issues/4319

### PR DESCRIPTION
**Pull request by macqbat**
# Related issues 🔗

Closes #4384 

# Description ℹ️

Add 'All' near by productType's in market selector

# Demo 📺

image
image

# Technical 👨‍🔧

Details of technical implementation that reviewers may need to be aware of, if applicable.

### CODE REVIEW

File: apps/trading/components/market-selector/product-selector.tsx [hunk 1]
```diff
@@ -11,11 +11,12 @@ export const Product = {
   Perpetual: 'Perpetual',
 } as const;
 
-export type ProductType = keyof typeof Product;
+export type ProductType = keyof typeof Product | 'All';
```
Comment (mattrussell36) [inline apps/trading/components/market-selector/product-selector.tsx:14]: Why create a union here? the const above is only for ui purposes so I'd just add a key to the ProductType const

**Review by mattrussell36 — CHANGES_REQUESTED** [review]

File: apps/trading/components/market-selector/product-selector.tsx (hunk 1 above)
Comment (macqbat) [reply apps/trading/components/market-selector/product-selector.tsx:14]: ok, there are a little bit misleading comment: `// Make sure these match the available __typename properties on product` 
Sure, gonna improve it

Comment (macqbat) [reply apps/trading/components/market-selector/product-selector.tsx]: done

File: apps/trading/components/market-selector/use-market-selector-list.ts [hunk 2]
```diff
@@ -28,7 +28,10 @@ export const useMarketSelectorList = ({
       .filter((m) => isMarketActive(m.state))
       // only selected product type
       .filter((m) => {
-        if (m.tradableInstrument.instrument.product.__typename === product) {
+        if (
+          product === 'All' ||
```
Comment (mattrussell36) [inline apps/trading/components/market-selector/use-market-selector-list.ts:32]:
```suggestion
          product === Product.All ||
```

**Review by mattrussell36 — CHANGES_REQUESTED** [review]
Can we tidy up the raw strings and use the product type constant plz

File: apps/trading/components/market-selector/market-selector.tsx [hunk 3]
```diff
@@ -39,7 +39,7 @@ export const MarketSelector = ({
 }) => {
   const [filter, setFilter] = useState<Filter>({
     searchTerm: '',
-    product: Product.Future,
+    product: 'All',
```
[the remark this answers is not in the document]
Comment (macqbat) [reply apps/trading/components/market-selector/market-selector.tsx]: done

File: apps/trading/components/market-selector/use-market-selector-list.ts (hunk 2 above)
Comment (macqbat) [reply apps/trading/components/market-selector/use-market-selector-list.ts]: Done

File: apps/trading/components/market-selector/use-market-selector-list.spec.tsx [hunk 4]
```diff
@@ -120,6 +120,13 @@ describe('useMarketSelectorList', () => {
       assets: [],
     });
     expect(result.current.markets).toEqual([markets[2]]);
+    rerender({
+      searchTerm: '',
+      product: 'All',
```
[the remark this answers is not in the document]
Comment (macqbat) [reply apps/trading/components/market-selector/use-market-selector-list.spec.tsx]: Done

**Review by mattrussell36 — APPROVED** [review]

**Pull request merged**